src/Discord/Internal/Types/ApplicationCommands.hs

```diff
@@ -0,0 +1,382 @@
+            ]
+      ]
+
+-- | Data type to be used when editing application commands. The specification
+-- is below. See `CreateApplicationCommand` for an explanation for the
+-- parameters.
+--
+-- https://discord.com/developers/docs/interactions/application-commands#edit-global-application-command
+data EditApplicationCommand = EditApplicationCommand
+  { editApplicationCommandName :: Maybe String,
+    editApplicationCommandDescription :: Maybe String,
+    editApplicationCommandOptions :: Maybe [ApplicationCommandOption],
+    editApplicationCommandDefaultPermission :: Maybe Bool,
+    editApplicationCommandType :: Maybe ApplicationCommandType
+  }
+
+instance Default EditApplicationCommand where
+  def = EditApplicationCommand Nothing Nothing Nothing Nothing Nothing
+
+instance ToJSON EditApplicationCommand where
+  toJSON EditApplicationCommand {..} =
+    object
+      [ (name, value)
+        | (name, Just value) <-
+            [ ("name", toJSON <$> editApplicationCommandName),
+              ("description", toJSON <$> editApplicationCommandDescription),
+              ("options", toJSON <$> editApplicationCommandOptions),
+              ("default_permission", toJSON <$> editApplicationCommandDefaultPermission),
+              ("type", toJSON <$> editApplicationCommandType)
+            ]
+      ]
+
+-- | The full information about an application command, obtainable with the
+-- various get requests. In theory, you never need to construct one of these -
+-- so if you are, reconsider what you're doing.
+--
+-- https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-structure
+data ApplicationCommand = ApplicationCommand
+  { -- | Unique id of the command.
+    applicationCommandId :: ApplicationCommandId,
+    -- | The type of the command.
+    applicationCommandType :: Maybe ApplicationCommandType,
+    -- | Unique id of the parent application (the bot).
+    applicationCommandApplicationId :: ApplicationId,
+    -- | The guild id of the command if not global.
+    applicationCommandGuildId :: Maybe GuildId,
+    -- | Must be 1-32 characters.
+    applicationCommandName :: String,
+    -- | Must be empty for USER and MESSAGE commands, otherwise 1-100 chars.
+    applicationCommandDescription :: String,
+    -- | CHAT_INPUT only, parameters to command
+    applicationCommandOptions :: Maybe [ApplicationCommandOption],
+    -- | whether the command is enabled by default when the app is added to a
+    -- guild. Defaults to true.
+    applicationCommandDefaultPermission :: Maybe Bool,
+    applicationCommandVersion :: Snowflake
+  }
+  deriving (Show)
+
+instance FromJSON ApplicationCommand where
+  parseJSON =
+    withObject
+      "ApplicationCommand"
+      ( \v ->
+          ApplicationCommand
+            <$> v .: "id"
+            <*> v .:? "type"
+            <*> v .: "application_id"
+            <*> v .:? "guild_id"
+            <*> v .: "name"
```

Comment:
Oh neat, this is a reasonable way to format https://github.com/aquarial/discord-haskell/issues/87. The `.:` v `.:?` formatting is still a minor complaint, but my main gripe is fixed here

Reply:
Other formatters may even fix this, but I've been focussing more on implementation tbh.

Reply:
:+1: 

Implementation is looking good